File: WBOR FINAL/PlayList.m
//
//  PlayList.m
//  WBOR FINAL
//
//  Created by Connor Smith on 12/17/11.
//  Copyright (c) 2011 __MyCompanyName__. All rights reserved.
//

#import "PlayList.h"

@implementation PlayList

@synthesize curSong, curArtist;

- (void)getCurrent{
    WBOR = [NSURL URLWithString:@"http://wbor-hr.appspot.com/updateinfo"];    
    NSData* data = [NSData dataWithContentsOfURL: 
                        WBOR];

    
    [self handleData:data];
}

- (void)handleData:(NSData *)responseData {
    //parse out the json data into a dictionary because it pretty much already is a dictionary
    NSError* error;
    NSDictionary* json = [NSJSONSerialization 
                          JSONObjectWithData:responseData 
                          
                          options:kNilOptions 
                          error:&error];
    
    NSString *song, *artist, *show;
    song    = [json objectForKey:@"song_string"];
    artist  = [json objectForKey:@"artist_string"];
    show    =[json objectForKey:@"program_title"];

    self.curShow = show;
    self.curSong = song;
    self.curArtist = artist;
}

@end

File: WBOR FINAL/Playlist.swift
//
//  Playlist.swift
//  WBOR
//
//  Created by Ruben Martinez Jr on 8/2/14.
//
//

import Foundation

class Playlist : NSObject {
    var wborInfo  : NSURL
    var curSong   : String?
    var curArtist : String?
    var curShow   : String?
    
    init(url: NSURL) {
        self.wborInfo = url
        super.init()
    }
    
    func getCurrent() {
        var data  = NSData(contentsOfURL: self.wborInfo)
        var error : NSError?
        var json  = NSJSONSerialization.JSONObjectWithData(data!,
                                                           options: nil,
                                                           error: &error) as Dictionary<String, String>

        self.curSong    = json["song_string"]!
        self.curArtist  = json["artist_string"]!
        self.curShow    = json["program_title"]!
        
    }
}